### PR TITLE
Expand supported options

### PR DESCRIPTION
**Pull request by aacunningham**
This is a bit of a refactor, but comes with some tests ~~(that can be manually run, workflows/tox to be added later)~~ (just kidding tox and CI stuff was added too), so hopefully should be safe enough.

This MR does a couple things:
1. As an internal refactor, the logic of the configuration has been pulled out into a new class, SentryDjangoConfig. This class is responsible for taking the configuration that user's provide via the `SENTRY` setting and handles all of the modifications to make it compatible with Sentry's init method. This also makes it incredibly easy to test this logic without dealing with Django's Apps.
2. As an external feature change, the `SENTRY` setting now supports the settings that we supported before, _along with_ all of the options that the Sentry init method supports. There isn't any validation that's done on the settings, so if it's something that couldn't be passed to Sentry, then it will fail the same way Sentry would.

Just with local checks, we're at almost 100% code coverage with the tests being added, which feels pretty good. The bits that are covered are the sections that I'd recommend deleting, specifically the git_sha file. But that's a task for another day.

### CODE REVIEW

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run Tox tests
+        id: test
+        uses: fedora-python/tox-github-action@master
+        with:
+          tox_env: ${{ matrix.tox_env }}
+    strategy:
+      matrix:
+        tox_env:
```
Comment (ghost) [inline .github/workflows/tox.yml:21]: Do we have to specify the `-django{version}` of each `tox_env`? I see that [in `django-freeipa-auth`](https://github.com/enervee/django-freeipa-auth/blob/main/.github/workflows/tox.yml) we only specify the `py{version}` part.

Comment (aacunningham) [reply .github/workflows/tox.yml:21]: I could maybe mess around with it, but in my tox.ini I've got the envlist:
```
envlist = {py36,py37,py38,py39}-django22,{py36,py37,py38,py39,py310}-django32,{py38,py39,py310}-django40
```

So I ran `tox -l` in the terminal and got the list:
```
py36-django22
py37-django22
py38-django22
py39-django22
py36-django32
py37-django32
py38-django32
py39-django32
py310-django32
py38-django40
py39-django40
py310-django40
```

And I just slapped that in here and called it a day? I think if I didn't do them separately like this, they would maybe get grouped together in the jobs. Like here, I've got 12 jobs, but maybe if I just called out the python versions, I'd only have 5 jobs and some of them would run multiple Django versions?

No clue how tox_env thingy works, maybe grouping them up would be better than running them all simultaneously.

Comment (ghost) [reply .github/workflows/tox.yml:21]: Oh interesting! Ok, maybe that's an update we can make to the github job in the `django-freeipa-auth` repo as well, that way we can separate those tests into separate jobs 👍

**Review by ghost — APPROVED** [review]
✅

**Review by rsbusby — APPROVED** [review]

**Pull request merged**